src/Microsoft.Health.Blob/Configs/BlobDataStoreConfiguration.cs

```diff
@@ -14,5 +14,10 @@ public class BlobDataStoreConfiguration
         public BlobDataStoreAuthenticationType AuthenticationType { get; set; } = BlobDataStoreAuthenticationType.ConnectionString;
 
         public BlobDataStoreRequestOptions RequestOptions { get; } = new BlobDataStoreRequestOptions();
+
+        /// <summary>
+        /// If set, the app ID of the user assigned managed identity to use when connecting to azure storage, if AUthenticationType == ManagedIdentity.
+        /// </summary>
+        public string UserAssignedManagedIdentityAppId { get; set; }
```

Comment:
Does it have to be a user assigned managed identity, or do system assigned also work here? Additionally, should we call this `ManagedIdentityClientId` to map with the parameter for the `ManagedIdentityCredential` type?

Reply:
Interesting the naming inconsistency between SQL and `ManagedIdentityCredential`.

Reply:
The parameter documentation in the SDK explicitly says user assigned:

// Parameters:
        //   clientId:
        //     The client id to authenticate for a user assigned managed identity. More information
        //     on user assigned managed identities can be found here: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-a-user-assigned-managed-identity-works-with-an-azure-vm
        //

Reply:
It defaults to null so not specifying it should switch back to system assigned.